### ISSUE TITLE
False positive about @UtilityClass and AbbreviationAsWordInName

### ISSUE THREAD

**Issue by LynnBroe** [checkstyle/checkstyle#12962]
I have read check documentation: https://checkstyle.org/config_naming.html#AbbreviationAsWordInName
I have downloaded the latest checkstyle from: https://checkstyle.org/cmdline.html#Download_and_Run
I have executed the cli and showed it below, as cli describes the problem better than 1,000 words

```bash
/var/tmp $ javac A.java


/var/tmp $ cat config.xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
          "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
          "https://checkstyle.org/dtds/configuration_1_3.dtd">
<module name="Checker">
  <module name="SuppressWarningsFilter"/>
  <property name="charset" value="UTF-8"/>
  <property name="severity" value="warning"/>
  <property name="fileExtensions" value="java, properties, xml"/>
  <module name="TreeWalker">
    <module name="AbbreviationAsWordInName">
      <property name="ignoreFinal" value="false"/>
      <property name="allowedAbbreviationLength" value="0"/>
      <property name="tokens"
                value="CLASS_DEF, INTERFACE_DEF, ENUM_DEF, ANNOTATION_DEF, ANNOTATION_FIELD_DEF,
                    PARAMETER_DEF, VARIABLE_DEF, METHOD_DEF, PATTERN_VARIABLE_DEF, RECORD_DEF,
                    RECORD_COMPONENT_DEF"/>
    </module>
  </module>
</module>

/var/tmp $ cat A.java
import lombok.experimental.UtilityClass;
@UtilityClass
public class A {
    private String varAA0 = "\u03bcs"; // violation

    private String varAA1 = "\u03bcs"; // violation
}
/var/tmp $ RUN_LOCALE="-Duser.language=en -Duser.country=US"
/var/tmp $ java $RUN_LOCALE -jar checkstyle-10.9.3-all.jar -c config.xml A.java
Starting audit...
[WARN] D:\MVN\newtest\src\main\java\.\A.java:6:20: Abbreviation in name 'varAA0' must contain no more than '1' consecutive capital letters. [AbbreviationAsWordInName]
[WARN] D:\MVN\newtest\src\main\java\.\A.java:8:20: Abbreviation in name 'varAA1' must contain no more than '1' consecutive capital letters. [AbbreviationAsWordInName]
Audit done.

However, the annotation @UtilityClass annotation from Lombok can transform the source code into the following code example:

public final class A {
    private static String varAA0 = "μs"; // violation
    private static String varAA1 = "μs"; // violation
    
    private A() {
        throw new java.lang.UnsupportedOperationException("This is a utility class and cannot be instantiated");
    }
}

/var/tmp $ java $RUN_LOCALE -jar checkstyle-10.9.3-all.jar -c config.xml gen/A.java
Starting audit...
Audit done.
```

---

**Describe what you expect in detail.**

---
The `A.java` and `gen\A.java` are equivalent. The `@UtilityClass` annotation will make the field `static`, so the analysis results are not same. Hence, I think it should be a false positive and the rule should consider the analysis results.

**Comment by romani** [comment]
ignoreStatic is true by default.
https://checkstyle.org/writingchecks.html#Limitations

Checkstyle is not aware of generated version of code.

You need to use some suppression, for example https://checkstyle.org/config_filters.html#SuppressionXpathSingleFilter and provide xpath that suppress all violations from this Check on classes with such annotation.

**Comment by romani** [comment]
workaround with filter 
```
$ cat config.xml 
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
          "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
          "https://checkstyle.org/dtds/configuration_1_3.dtd">
<module name="Checker">
  <module name="SuppressWarningsFilter"/>
  <property name="charset" value="UTF-8"/>
  <property name="severity" value="warning"/>
  <property name="fileExtensions" value="java, properties, xml"/>
  <module name="TreeWalker">
    <module name="AbbreviationAsWordInName">
      <property name="ignoreFinal" value="false"/>
      <property name="allowedAbbreviationLength" value="0"/>
      <property name="tokens"
                value="CLASS_DEF, INTERFACE_DEF, ENUM_DEF, ANNOTATION_DEF, ANNOTATION_FIELD_DEF,
                    PARAMETER_DEF, VARIABLE_DEF, METHOD_DEF, PATTERN_VARIABLE_DEF, RECORD_DEF,
                    RECORD_COMPONENT_DEF"/>
    </module>
    <module name="SuppressionXpathSingleFilter">
      <property name="checks" value="AbbreviationAsWordInName"/>
      <property name="query" value="//CLASS_DEF[.//ANNOTATION/IDENT[@text='UtilityClass']]/OBJBLOCK/VARIABLE_DEF/IDENT"/>
    </module>
  </module>
</module>


$ cat Test.java 
import lombok.experimental.UtilityClass;
@UtilityClass
public class A {
    private String varAA0 = "\u03bcs"; // filtered violation

    private String varAA1 = "\u03bcs"; // filtered violation

    void foo() {
      String varAA1 = "\u03bcs"; // violation
    }
}

$ java -jar checkstyle-10.24.0-all.jar -c config.xml Test.java
Starting audit...
[WARN] /var/tmp/Test.java:9:14: Abbreviation in name 'varAA1' must contain no more than '1' consecutive 
     capital letters. [AbbreviationAsWordInName]
Audit done.

```
